src/cmd-prune-containers

```diff
@@ -0,0 +1,168 @@
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(prog="coreos-assembler prune-containers")
+    parser.add_argument("--policy", required=True, type=str, help="Path to policy YAML file")
+    parser.add_argument("--dry-run", help="Don't actually delete anything", action='store_true')
+    parser.add_argument("-v", help="Increase verbosity", action='store_true')
+    parser.add_argument("--registry-auth-file", default=os.environ.get("REGISTRY_AUTHFILE"), help="Path to AWS config file")
+    parser.add_argument("--stream", type=str, help="CoreOS stream", required=True, choices=STREAMS.keys())
+    parser.add_argument("repository_url", help="container images URL")
+    return parser.parse_args()
+
+
+def convert_to_days(duration_arg):
+
+    days = 0
+    match = regexp.match(r'^([0-9]+)([dDmMyYwW])$', duration_arg)
+
+    if match is None:
+        raise ValueError(f"Incorrect duration '{duration_arg}'. Valid values are in the form of 1d, 2w, 3m, 4y")
+
+    unit = match.group(2)
+    value = int(match.group(1))
+    match unit:
+        case "y" | "Y":
+            days = value * 365
+        case "m" | "M":
+            days = value * 30
+        case "w" | "W":
+            days = value * 7
+        case "d" | "D":
+            days = value
+        case _:
+            raise ValueError(f"Invalid unit '{match.group(2)}'. Please use y (years), m (months), w (weeks), or d (days).")
+
+    return days
+
+
+def skopeo_delete(repo, image, auth):
+
+    skopeo_args = ["skopeo", "delete", f"docker://{repo}:{image}"]
+    if auth is not None:
+        skopeo_args.append(f"--authfile {auth}")
+
+    subprocess.run(skopeo_args)
+
+
+# FIXME : move to cosa_lib
+# https://github.com/coreos/coreos-assembler/pull/3798/files#r1673481990
+
+def parse_fcos_version(version):
+    m = regexp.match(r'^([0-9]{2})\.([0-9]{8})\.([0-9]+)\.([0-9]+)$', version)
+    if m is None:
+        raise ValueError(f"Incorrect versioning for FCOS build {version}")
+    try:
+        timestamp = datetime.datetime.strptime(m.group(2), '%Y%m%d')
+    except ValueError:
+        raise Exception(f"FCOS build {version} has incorrect date format. It should be in (%Y%m%d)")
+    return (timestamp, int(m.group(3)))
+
+
+def get_update_graph(stream):
+
+    url = f"https://raw.githubusercontent.com/coreos/fedora-coreos-streams/main/updates/{stream}.json"
+    r = requests.get(url)
+    if r.status_code != 200:
+        raise Exception(f"Could not download update graph for {stream}. HTTP {r.status_code}")
+    return r.json()
+
+
+class BarrierRelease(Exception):
+    pass
+
+
+def main():
+
+    args = parse_args()
+
+    # Load the policy file
+    with open(args.policy, "r") as f:
+        policy = yaml.safe_load(f)
+    if args.stream not in policy:
+        raise Exception(f"Stream {args.stream} is not defined in policy file.")
+    if 'containers' not in policy[args.stream]:
+        raise Exception(f"No containers section for {args.stream} stream in policy.")
```

Comment:
I feel like those should just be no-ops instead of hard errors.

Reply:
good point

Reply:
Yeah, sorry but no op I mean just `return`.